Dockerfile

```diff
@@ -12,19 +12,7 @@ LABEL org.opencontainers.image.ref.name="jnovack/autossh" \
       org.opencontainers.image.revision=$COMMIT \
       org.opencontainers.image.url="https://hub.docker.com/r/jnovack/autossh/"
 
-ENTRYPOINT ["/entrypoint.sh"]
```

Comment:
This script IS an [`ENTRYPOINT`](https://docs.docker.com/engine/reference/builder/#entrypoint) and is using the **exec** form intentionally.

There is no other application (that should be) run in this container.  There's no other reason to fire up this container other than this application, thus `ENTRYPOINT` is the proper argument. 

[We are not overwriting any variables from the command-line](https://goinbigdata.com/docker-run-vs-cmd-vs-entrypoint/) and `ENTRYPOINT` provides us a mechanism to run the container AS an executable (using `docker run`).

Please revert this change.

Reply:
I'm holding on this.

> `ENTRYPOINT` provides us a mechanism to run the container AS an executable (using `docker run`).
> There is no other application (that should be) run in this container. There's no other reason to fire up this container other than this application, thus ENTRYPOINT is the proper argument.

Using `CMD` does not prevent or hinder usage of `docker run` or suggest that an container is multi-application. That distinction wouldn't make sense, as the entire container paradigm expects all containers to be single application. Most of Docker's own library images use `CMD`, such as [nginx](https://github.com/nginxinc/docker-nginx/blob/master/stable/alpine/Dockerfile)

`ENTRYPOINT` is meant for containers that wrap and package a single binary, and provide a means for users to pass arguments to that binary without having to re-specify the binary itself. The [StackOverflow link](https://stackoverflow.com/a/21564990) I posted has a good explanation of what the use case of this is. Because the `entrypoint.sh` does not accept any flags or arguments from the command line, there is no reason to use `ENTRYPOINT` here. (And notably, this is why I rename the `entrypoint.sh` to `init`, to avoid this word mangling)

More importantly, it prevents the following issue:

```
docker run -e SSH_TUNNEL_HOST=192.168.1.1 -e SSH_TUNNEL_LOCAL=8080 -v ./mysshkey:/id_rsa jnovack/autossh
```
Will work with both `CMD` and `ENTRYPOINT` notation, where the specified default `CMD` will mean that the `entrypoint.sh` script is used

However, if the user wants to manually override script to set monitor port:
```
docker run -v ./mysshkey:/id_rsa jnovack/autossh autossh -M 12345 -i /id_rsa -R 8080:localhost:80
```
Then this will not work in `ENTRYPOINT` as Docker will try to pass all the overridden command as arguments to `entrypoint.sh`, whereas the `CMD` would just be ignored and the user's command would be executed